src/spend.rs

```diff
@@ -497,6 +502,40 @@ fn derived_desc(
     desc.derive(coin.deriv_index, secp)
 }
 
+/// Get value to use for transaction nLockTime in order to
+/// discourage fee sniping.
+///
+/// The approach follows that taken by Bitcoin Core:
+/// - most of the time, the value returned will be the current
+/// block height, but will randomly be up to 100 blocks earlier.
+/// - if the current tip is more than [`MAX_ANTI_FEE_SNIPING_TIP_AGE_SECS`]
+/// seconds old, a locktime value of 0 will be returned.
+pub fn anti_fee_sniping_locktime(
+    now: Duration,
+    tip_height: u32,
+    tip_time_secs: Option<u32>,
+) -> LockTime {
+    tip_time_secs
+        .map(|tip_time| now.as_secs().saturating_sub(tip_time.into()))
+        .filter(|tip_age| *tip_age <= MAX_ANTI_FEE_SNIPING_TIP_AGE_SECS)
+        .map(|_| {
+            // Randomly (approx 10% of cases) set locktime further back
+            // using current time as source of randomness.
+            let nanos = now.subsec_nanos();
+            // Note this condition will fail if nano precision is not available
+            // and so nothing will be subtracted.
+            let delta = if nanos % 10 == 1 {
+                (nanos % 1000) / 10 + 1 // a number in [1, 100]
```

Comment:
Why `% 1000 / 10` and not `% 100`?

Reply:
This ensures we can get all values in `[1, 100]`. Given the condition `nanos % 10 == 1`, using `% 100` would limit the set of possible values.

Reply:
Ugh, of course! Thanks.